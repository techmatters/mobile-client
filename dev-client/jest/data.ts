/*
 * Copyright Â© 2024 Technology Matters
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

import {AppState} from 'terraso-mobile-client/store';

export const testState: Partial<AppState> = {
  account: {
    currentUser: {
      data: {
        id: '1',
        email: 'test@domain.com',
        firstName: 'firstname',
        lastName: 'lastname',
        profileImage: '',
        preferences: {},
      },
      fetching: false,
    },
    users: {
      '1': {
        id: '1',
        email: 'test@domain.com',
        firstName: 'firstname',
        lastName: 'lastname',
        profileImage: '',
        preferences: {},
      },
    },
  } as any,
  site: {
    sites: {
      '1': {
        id: '1',
        name: '1',
        latitude: 1,
        longitude: 1,
        elevation: 1,
        privacy: 'PRIVATE',
        archived: false,
        updatedAt: '',
        notes: {},
        projectId: '1',
      },
    },
  },
  project: {
    projects: {
      '1': {
        id: '1',
        name: '1',
        privacy: 'PRIVATE',
        archived: false,
        updatedAt: '',
        measurementUnits: 'METRIC',
        description: 'desc',
        memberships: {},
        sites: {'1': true},
      },
    },
  },
};
