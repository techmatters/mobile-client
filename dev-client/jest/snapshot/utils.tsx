/*
 * Copyright Â© 2023 Technology Matters
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */
import {useRef} from 'react';
import {Portal} from 'react-native-paper';
import {Provider} from 'react-redux';

import {BottomSheetModalProvider} from '@gorhom/bottom-sheet';
import {NavigationContainer} from '@react-navigation/native';
import {render as rnRender} from '@testing-library/react-native';
import {NativeBaseProvider} from 'native-base';

import {
  RootStack,
  RootStackParamList,
} from 'terraso-mobile-client/navigation/types';
import {AppState, createStore} from 'terraso-mobile-client/store';
import {theme} from 'terraso-mobile-client/theme';

// NativeBase: https://docs.nativebase.io/testing
const nativeBaseInset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

type WrapperProps = {
  initialState?: Partial<AppState>;
  route?: keyof RootStackParamList;
};

const TestWrapper = ({
  initialState,
  route,
  children,
}: React.PropsWithChildren<WrapperProps>) => {
  const store = useRef(createStore(initialState));
  return (
    <Provider store={store.current}>
      <NavigationContainer>
        <NativeBaseProvider
          theme={theme}
          initialWindowMetrics={nativeBaseInset}>
          <BottomSheetModalProvider>
            <Portal.Host>
              {route && (
                <RootStack.Navigator>
                  <RootStack.Screen name={route}>
                    {() => children}
                  </RootStack.Screen>
                </RootStack.Navigator>
              )}
              {!route && children}
            </Portal.Host>
          </BottomSheetModalProvider>
        </NativeBaseProvider>
      </NavigationContainer>
    </Provider>
  );
};

export const render = (
  ui: React.ReactElement,
  wrapperProps: WrapperProps = {},
) => {
  const Wrapper = ({children}: React.PropsWithChildren) => (
    <TestWrapper {...wrapperProps}>{children}</TestWrapper>
  );

  return rnRender(ui, {wrapper: Wrapper});
};
