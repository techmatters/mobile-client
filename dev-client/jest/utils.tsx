/*
 * Copyright Â© ${YEAR} Technology Matters
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */
import {render} from '@testing-library/react-native';
import {NativeBaseProvider} from 'native-base';
import {theme} from 'terraso-mobile-client/theme';

// NativeBase: https://docs.nativebase.io/testing
const nativeBaseInset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

const TestWrapper = ({children}: React.PropsWithChildren) => {
  return (
    <NativeBaseProvider theme={theme} initialWindowMetrics={nativeBaseInset}>
      {children}
    </NativeBaseProvider>
  );
};

export const customRender: typeof render = (ui, options) => {
  return render(ui, {wrapper: TestWrapper, ...options});
};
