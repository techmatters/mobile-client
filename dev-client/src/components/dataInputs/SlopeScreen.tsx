import {Text} from 'native-base';

export const SlopeScreen = (_: {siteId: string}) => {
  return <Text margin="auto">Unimplemented slope screen</Text>;
};
