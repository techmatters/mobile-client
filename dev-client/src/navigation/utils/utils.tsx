/*
 * Copyright Â© 2024 Technology Matters
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

import {
  ParamListBase,
  TypedNavigator,
  useNavigation as useNavigationNative,
} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';

import {ScreenDefinitions} from 'terraso-mobile-client/navigation/types';

export const generateScreens = <T extends TypedNavigator<any, any>>(
  Navigator: T,
  definitions: ScreenDefinitions,
) =>
  Object.entries(definitions).map(([name, Screen]) => (
    <Navigator.Screen
      name={name}
      key={name}
      children={props => <Screen {...((props.route.params ?? {}) as any)} />}
    />
  ));

export const createNavigationHook =
  <ParamList extends ParamListBase>() =>
  <Name extends keyof ParamList>() =>
    useNavigationNative<NativeStackNavigationProp<ParamList, Name>>();
