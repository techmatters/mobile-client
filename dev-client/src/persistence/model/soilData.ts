export type LocalSoilData = {
  bedrock?: null | number;
  crossSlope?: null | string;
  depthDependentData: LocalDepthDependentSoilData[];
  depthIntervalPreset: string;
  depthIntervals: LocalDepthInterval[];
  downSlope?: null | string;
  floodingSelect?: null | string;
  grazingSelect?: null | string;
  landCoverSelect?: null | string;
  limeRequirementsSelect?: null | string;
  slopeAspect?: null | number;
  slopeLandscapePosition?: null | string;
  slopeSteepnessDegree?: null | number;
  slopeSteepnessPercent?: null | number;
  slopeSteepnessSelect?: null | string;
  soilDepthSelect?: null | string;
  surfaceCracksSelect?: null | string;
  surfaceSaltSelect?: null | string;
  surfaceStoninessSelect?: null | string;
  waterTableDepthSelect?: null | string;
};

export type LocalDepthDependentSoilData = {
  carbonates?: null | string;
  clayPercent?: null | number;
  colorChroma?: null | number;
  colorHue?: null | number;
  colorPhotoLightingCondition?: null | string;
  colorPhotoSoilCondition?: null | string;
  colorPhotoUsed?: null | boolean;
  colorValue?: null | number;
  conductivity?: null | any;
  conductivityTest?: null | string;
  conductivityUnit?: null | string;
  depthInterval: LocalDepthInterval;
  ph?: null | any;
  phTestingMethod?: null | string;
  phTestingSolution?: null | string;
  rockFragmentVolume?: null | string;
  sodiumAbsorptionRatio?: null | any;
  soilOrganicCarbon?: null | any;
  soilOrganicCarbonTesting?: null | string;
  soilOrganicMatter?: null | any;
  soilOrganicMatterTesting?: null | string;
  structure?: null | string;
  texture?: null | string;
};

export type LocalDepthInterval = {
  carbonatesEnabled?: null | boolean;
  depthInterval: LocalDepthInterval;
  electricalConductivityEnabled?: null | boolean;
  label: string;
  phEnabled?: null | boolean;
  sodiumAdsorptionRatioEnabled?: null | boolean;
  soilColorEnabled?: null | boolean;
  soilOrganicCarbonMatterEnabled?: null | boolean;
  soilStructureEnabled?: null | boolean;
  soilTextureEnabled?: null | boolean;
};
