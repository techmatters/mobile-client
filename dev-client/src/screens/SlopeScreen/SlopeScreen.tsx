/*
 * Copyright Â© 2023 Technology Matters
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */
import {useCallback, useRef} from 'react';
import {useTranslation} from 'react-i18next';
import {useSelector} from 'terraso-mobile-client/store';
import {useNavigation} from 'terraso-mobile-client/navigation/hooks/useNavigation';
import {DataInputSummary} from 'terraso-mobile-client/components/DataInputSummary';
import {
  renderShape,
  renderSteepness,
} from 'terraso-mobile-client/screens/SlopeScreen/utils/renderValues';
import {
  Column,
  Row,
  Heading,
} from 'terraso-mobile-client/components/NativeBaseAdapters';
import {
  selectSoilData,
  useSiteProjectSoilSettings,
} from 'terraso-client-shared/selectors';
import {IconButton} from 'terraso-mobile-client/components/Icons';
import {BottomSheetModal} from '@gorhom/bottom-sheet';
import {BasicInfoModal} from 'terraso-mobile-client/components/infoModals/BasicInfoModal';
import {SlopeInfoContent} from 'terraso-mobile-client/screens/SlopeScreen/components/SlopeInfoContent';

export const SlopeScreen = ({siteId}: {siteId: string}) => {
  const {t} = useTranslation();
  const soilData = useSelector(selectSoilData(siteId));
  const required = useSiteProjectSoilSettings(siteId)?.slopeRequired ?? false;
  const navigation = useNavigation();

  const infoBottomSheetRef = useRef<BottomSheetModal>(null);
  const onInfoPress = useCallback(
    () => infoBottomSheetRef.current?.present(),
    [infoBottomSheetRef],
  );
  const onInfoClose = useCallback(
    () => infoBottomSheetRef.current?.dismiss(),
    [infoBottomSheetRef],
  );

  const steepnessValue = renderSteepness(t, soilData);
  const shapeValue = renderShape(t, soilData);

  const onSteepness = useCallback(
    () => navigation.push('SLOPE_STEEPNESS', {siteId}),
    [navigation, siteId],
  );

  const onShape = useCallback(
    () => navigation.push('SLOPE_SHAPE', {siteId}),
    [navigation, siteId],
  );

  return (
    <>
      <Column space="1px">
        <Row backgroundColor="primary.contrast" p="15px" alignItems="center">
          <Heading variant="h6">{t('slope.title')}</Heading>
          <IconButton
            name="info"
            onPress={onInfoPress}
            size="md"
            _icon={{color: 'primary'}}
          />
        </Row>
        <DataInputSummary
          required={required}
          complete={steepnessValue !== undefined}
          label={t('slope.steepness.short_title').toLocaleUpperCase()}
          value={steepnessValue}
          onPress={onSteepness}
        />
        <DataInputSummary
          required={required}
          complete={shapeValue !== undefined}
          label={t('slope.shape.title').toLocaleUpperCase()}
          value={shapeValue}
          onPress={onShape}
        />
      </Column>
      <BasicInfoModal ref={infoBottomSheetRef} onClose={onInfoClose}>
        <SlopeInfoContent />
      </BasicInfoModal>
    </>
  );
};
