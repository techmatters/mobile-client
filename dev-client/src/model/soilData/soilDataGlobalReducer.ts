/*
 * Copyright Â© 2024 Technology Matters
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

import {setUsers} from 'terraso-client-shared/account/accountSlice';
import * as soilDataService from 'terraso-client-shared/soilId/soilDataService';
import {createAsyncThunk} from 'terraso-client-shared/store/utils';

import {setProjects} from 'terraso-mobile-client/model/project/projectSlice';
import {setSites} from 'terraso-mobile-client/model/site/siteSlice';
import {
  setProjectSettings,
  setSoilData,
  updateSoilIdStatus,
} from 'terraso-mobile-client/model/soilData/soilDataSlice';
import {setSoilMetadata} from 'terraso-mobile-client/model/soilMetadata/soilMetadataSlice';
import {createGlobalReducer} from 'terraso-mobile-client/store/reducers';

export const fetchSoilDataForUser = createAsyncThunk(
  'soilId/fetchSoilDataForUser',
  soilDataService.fetchSoilDataForUser,
);

export const soilIdGlobalReducer = createGlobalReducer(builder => {
  builder.addCase(fetchSoilDataForUser.fulfilled, (state, {payload}) => {
    setProjects(state.project, payload.projects);
    setSites(state.site, payload.sites);
    setUsers(state.account, payload.users);
    setProjectSettings(state.soilData, payload.projectSoilSettings);
    setSoilData(state.soilData, payload.soilData);
    setSoilMetadata(state.soilMetadata, payload.soilMetadata);
    updateSoilIdStatus(state.soilData, 'ready');
  });

  builder.addCase(fetchSoilDataForUser.pending, state => {
    updateSoilIdStatus(state.soilData, 'loading');
  });

  builder.addCase(fetchSoilDataForUser.rejected, state => {
    updateSoilIdStatus(state.soilData, 'error');
  });
});
