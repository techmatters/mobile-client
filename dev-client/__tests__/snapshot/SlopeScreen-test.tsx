/*
 * Copyright Â© 2024 Technology Matters
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

import {testState} from '@testing/integration/data';
import {render} from '@testing/integration/utils';

import {collectionMethods} from 'terraso-client-shared/soilId/soilIdTypes';
import {fromEntries} from 'terraso-client-shared/utils';

import {methodRequired} from 'terraso-mobile-client/model/soilData/soilDataSlice';
import {SlopeScreen} from 'terraso-mobile-client/screens/SlopeScreen/SlopeScreen';

test('renders correctly', () => {
  const screen = render(<SlopeScreen siteId="1" />, {
    route: 'SITE_TABS',
    initialState: {
      ...testState,
      soilData: {
        status: 'ready',
        soilData: {
          '1': {
            depthIntervalPreset: 'NRCS',
            depthIntervals: [],
            depthDependentData: [],
            slopeSteepnessSelect: 'FLAT',
          },
        },
        soilSync: {},
        projectSettings: {
          '1': {
            ...fromEntries(
              collectionMethods.map(method => [methodRequired(method), false]),
            ),
            soilPitRequired: false,
            depthIntervalPreset: 'NRCS',
            depthIntervals: [],
            slopeRequired: true,
          },
        },
      },
    },
  }).toJSON();

  expect(screen).toMatchSnapshot();
});
