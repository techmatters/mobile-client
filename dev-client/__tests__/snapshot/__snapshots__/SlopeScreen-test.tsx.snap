// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          fullScreenSwipeShadowEnabled={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          hasLargeHeader={false}
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onGestureCancel={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillAppear={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          stackPresentation="push"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              collapsable={false}
              style={
                [
                  {
                    "flex": 1,
                  },
                  {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <RCTScrollView
                contentContainerStyle={
                  [
                    {},
                    {
                      "dataSet": {},
                    },
                    {},
                  ]
                }
                dataSet={{}}
                style={
                  {
                    "backgroundColor": "#E0E0E0",
                  }
                }
              >
                <View>
                  <View
                    style={
                      [
                        {
                          "flexDirection": "row",
                        },
                        {
                          "alignItems": "center",
                          "backgroundColor": "#FFFFFF",
                          "padding": 15,
                        },
                      ]
                    }
                  >
                    <Text
                      lineHeight="32px"
                      style={
                        {
                          "color": "#1A202C",
                          "fontSize": 20,
                          "fontWeight": "500",
                        }
                      }
                    >
                      Slope
                    </Text>
                    <View
                      style={
                        {
                          "marginLeft": 6,
                        }
                      }
                    >
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                      >
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                        >
                          <Icon
                            name="info"
                            size={20}
                            style={
                              [
                                {
                                  "textAlign": "center",
                                  "verticalAlign": "middle",
                                },
                                {
                                  "borderRadius": 100,
                                  "padding": 4,
                                },
                                {
                                  "backgroundColor": "#00000000",
                                  "color": "#001923B3",
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <RCTScrollView
                        focusHook={[Function]}
                      >
                        <View>
                          <View
                            style={
                              [
                                {
                                  "flexDirection": "column",
                                },
                                {
                                  "padding": 16,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                [
                                  {
                                    "flexDirection": "row",
                                  },
                                  {
                                    "alignItems": "center",
                                    "marginBottom": 16,
                                  },
                                ]
                              }
                            >
                              <Text
                                lineHeight="32px"
                                style={
                                  {
                                    "color": "#1A202C",
                                    "fontSize": 24,
                                    "fontWeight": "700",
                                  }
                                }
                              >
                                Slope
                              </Text>
                              <View
                                style={
                                  {
                                    "flex": 1,
                                  }
                                }
                              />
                              <View
                                accessibilityRole="button"
                                accessibilityState={
                                  {
                                    "busy": undefined,
                                    "checked": undefined,
                                    "disabled": undefined,
                                    "expanded": undefined,
                                    "selected": undefined,
                                  }
                                }
                                accessibilityValue={
                                  {
                                    "max": undefined,
                                    "min": undefined,
                                    "now": undefined,
                                    "text": undefined,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                              >
                                <Icon
                                  name="close"
                                  size={24}
                                  style={
                                    [
                                      {
                                        "textAlign": "center",
                                        "verticalAlign": "middle",
                                      },
                                      {
                                        "borderRadius": 100,
                                        "padding": 12,
                                      },
                                      {
                                        "backgroundColor": "#EEEEEE",
                                        "color": "#001923B3",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <Text
                              letterSpacing="0.15px"
                              lineHeight="24px"
                              style={
                                {
                                  "color": "#1A202C",
                                  "fontSize": 16,
                                  "fontWeight": "400",
                                }
                              }
                            >
                              Describe the characteristics of the slope of the land on which the site is located.

Instructions on how to collect each data input are available by tapping on the input name.

Slope is important in identifying the soil. It also helps determine the soil’s erosion potential, ability to hold water, and more.
                            </Text>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                  <View
                    style={
                      [
                        {
                          "backgroundColor": "rgba(202, 196, 208, 1)",
                          "height": 0.5,
                        },
                        undefined,
                        false,
                        false,
                        undefined,
                      ]
                    }
                  />
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <View
                      style={
                        [
                          {
                            "flexDirection": "row",
                          },
                          {
                            "backgroundColor": "#E2FFE5",
                            "padding": 15,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          {
                            "width": 37,
                          }
                        }
                      >
                        <Icon
                          color="#00582B"
                          name="check-circle"
                          size={24}
                          style={
                            {
                              "marginTop": -3,
                            }
                          }
                        />
                      </View>
                      <Text
                        letterSpacing="0.17px"
                        lineHeight="20px"
                        style={
                          {
                            "color": "#1A202C",
                            "fontSize": 14,
                            "fontWeight": "700",
                            "textTransform": "uppercase",
                          }
                        }
                      >
                        Steepness
                      </Text>
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      />
                      <Text
                        letterSpacing="0.17px"
                        lineHeight="20px"
                        style={
                          {
                            "color": "#1A202C",
                            "fontSize": 14,
                            "fontWeight": "400",
                          }
                        }
                      >
                        Flat 0–2%
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      [
                        {
                          "backgroundColor": "rgba(202, 196, 208, 1)",
                          "height": 0.5,
                        },
                        undefined,
                        false,
                        false,
                        undefined,
                      ]
                    }
                  />
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <View
                      style={
                        [
                          {
                            "flexDirection": "row",
                          },
                          {
                            "backgroundColor": "#FFFFFF",
                            "padding": 15,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          {
                            "width": 37,
                          }
                        }
                      >
                        <Icon
                          name="radio-button-unchecked"
                          size={24}
                          style={
                            {
                              "marginTop": -3,
                            }
                          }
                        />
                      </View>
                      <Text
                        letterSpacing="0.17px"
                        lineHeight="20px"
                        style={
                          {
                            "color": "#1A202C",
                            "fontSize": 14,
                            "fontWeight": "700",
                            "textTransform": "uppercase",
                          }
                        }
                      >
                        Shape
                      </Text>
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backTitleVisible={true}
            backgroundColor="rgb(255, 255, 255)"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={false}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="LOCATION_DASHBOARD"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={false}
          />
        </RNSScreen>
      </RNSScreenStack>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
